lib/source-map-resolve-node.js

```diff
@@ -70,9 +70,24 @@ function resolveSourceMapSync(code, codeUrl, read) {
   return mapData
 }
 
-var dataUriRegex = /^data:([^,;]*)(;[^,;]*)*(?:,(.*))?$/
+var dataUriRegex = /^data:(([^,;]*)(;[^,;]*)*)(?:,(.*))?$/
 var jsonMimeTypeRegex = /^(?:application|text)\/json$/
 
+function decodeBase64String(utfLabel, b64) {
+  var buf = base64ToBuf(b64);
+  var txt = new TextDecoder(utfLabel, {fatal: true}).decode(buf);
+  return txt; 
+  function base64ToBuf(str) {
```

Comment:
1. I think this would be nicer:

   ```js
   return new TextDecoder(utfLabel, {fatal: true}).decode(base64ToBuf(b64))
   ```

2. What is the motivation for `fatal: true`? I don’t know what’s correct myself, so I’m asking :)

3. Can you move `base64ToBuf` to a toplevel function?

4. We need to check if `TextDecoder` and `Uint8Array` exist and fall back to just `atob` if they don’t for backwards compatibility.